signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/SignalFxReporter.java

```diff
@@ -38,25 +39,28 @@
     private final Set<MetricDetails> detailsToAdd;
     private final MetricMetadata metricMetadata;
     private final boolean useLocalTime;
+    private final ImmutableMap<String,String> defaultDimensions;
 
     protected SignalFxReporter(MetricRegistry registry, String name, MetricFilter filter,
                                  TimeUnit rateUnit, TimeUnit durationUnit,
                                  AggregateMetricSender aggregateMetricSender,
                                  Set<MetricDetails> detailsToAdd,
                                  MetricMetadata metricMetadata) {
-        this(registry, name, filter, rateUnit, durationUnit, aggregateMetricSender, detailsToAdd, metricMetadata, false);
+        this(registry, name, filter, rateUnit, durationUnit, aggregateMetricSender, detailsToAdd,
+                metricMetadata, false, ImmutableMap.<String, String>of());
     }
 
     public SignalFxReporter(MetricRegistry registry, String name, MetricFilter filter,
                               TimeUnit rateUnit, TimeUnit durationUnit,
                               AggregateMetricSender aggregateMetricSender,
                               Set<MetricDetails> detailsToAdd, MetricMetadata metricMetadata,
-                              boolean useLocalTime) {
+                              boolean useLocalTime, ImmutableMap<String,String> defaultDimensions) {
```

Comment:
Okay I will make it a copy on input so that people can't change things after the constructor is called (even if by accident)